cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/core/service/worker/impl/CoordinateCleanerImpl.java

```diff
@@ -0,0 +1,122 @@
+import edu.asu.diging.cord19.explorer.core.model.task.Task;
+import edu.asu.diging.cord19.explorer.core.model.task.TaskStatus;
+import edu.asu.diging.cord19.explorer.core.model.task.impl.TaskImpl;
+import edu.asu.diging.cord19.explorer.core.mongo.PublicationRepository;
+import edu.asu.diging.cord19.explorer.core.service.worker.CoordinateCleaner;
+import java.util.List;
+
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.data.util.CloseableIterator;
+
+import edu.asu.diging.cord19.explorer.core.model.Publication;
+import edu.asu.diging.cord19.explorer.core.model.impl.CountriesImpl;
+import edu.asu.diging.cord19.explorer.core.model.impl.MapTotalsImpl;
+import edu.asu.diging.cord19.explorer.core.model.impl.PersonImpl;
+import edu.asu.diging.cord19.explorer.core.model.impl.PublicationImpl;
+import edu.asu.diging.cord19.explorer.core.model.impl.WikipediaArticleImpl;
+
+@Service
+@PropertySource({ "classpath:config.properties", "${appConfigFile:classpath:}/app.properties" })
+public class CoordinateCleanerImpl implements CoordinateCleaner {
+
+    @Autowired
+    private TaskRepository taskRepo;
+
+    @Autowired
+    private PublicationRepository pubRepo;
+
+    @Autowired
+    private MongoTemplate mongoTemplate;
+
+    @Autowired
+    private MapTotalsRepository totalsRepo;
+    
+
+    public void calculateCountryStats() {
+
+        MapTotalsImpl totals = new MapTotalsImpl();
+        try (CloseableIterator<CountriesImpl> countries = mongoTemplate.stream(new Query().noCursorTimeout(),
+                CountriesImpl.class)) {
+            while (countries.hasNext()) {
+                CountriesImpl country = countries.next();
+                if (country.getProperties().getSelectedWikipediaCount() > 0) {
+                    if (country.getProperties().getSelectedWikipediaCount() > totals.getHighCount()) {
+                        totals.setHighCount(country.getProperties().getSelectedWikipediaCount());
+                    }
+                    if (country.getProperties().getSelectedWikipediaCount() < totals.getLowCount()) {
+                        totals.setLowCount(country.getProperties().getSelectedWikipediaCount());
+                    }
+                }
+            }
+        }
+
+        totalsRepo.save(totals);
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see edu.asu.diging.cord19.explorer.core.service.worker.impl.DocumentCleaner#
+     *
+     */
+    @Override
+    @Async
+    public void startCleaningCoordinates(String taskId) {
+
+        Optional<TaskImpl> optional = taskRepo.findById(taskId);
+        if (!optional.isPresent()) {
+            return;
+        }
+
+        Task task = optional.get();
+        task.setStatus(TaskStatus.PROCESSING);
+        taskRepo.save((TaskImpl) task);
+        
+        CoordinateParser coordinateParser = new CoordinateParser();
+
+        try (CloseableIterator<PublicationImpl> docs = mongoTemplate.stream(new Query(), PublicationImpl.class)) {
```

Comment:
i'm fairly certain you need to add `noCursorTimeout` here since there are a lot of publications.